### ISSUE TITLE
How to validate optional parameters?

### ISSUE THREAD

**Issue by jezzsantos** [markusbohl/fluent-ts-validator#2]
Lets say I have a DTO like this:

```
class Dto{
  property1?:number
}
```

and a validator:

```
class MyValidator extends AbstractValidator<Dto>{
}
```

How would I write a rule that only validates the number if it is defined, since it is optional?

I tried:
```
this.validateIfNumber(dto => dto.property1)
            .isPositive()
            .whenDefined()
            .withFailureMessage(`must be a number above zero`)

```

But compiler obviously complains that: `Type 'number | undefined' is not assignable to type 'number'` because `validateIfNumber()` must have a `number` not a `number | undefined`

**Comment by markusbohl** [comment]
Oh, I did not recognize this compiler warning before. Thanks for pointing that out.

Well, your code is fine. And you will be able to compile it when disabling `strict` and/or `strictNullChecks` options in your `tsconfig.json` (under `compilerOptions`).

Obviously, this is not exactly nice. Let me think about a proper solution.

**Comment by markusbohl** [comment]
Just released version [2.1.0](https://github.com/markusbohl/fluent-ts-validator/releases/tag/2.1.0) which allows you to keep your compiler options strict and define validation rules for optional properties.

**Comment by jezzsantos** [comment]
thnx
